### DISCUSSION TITLE
pointerover/out events don't fire when loading with a mouse in viewport

### DISCUSSION THREAD

**Discussion by batman4444** [pixijs/pixijs#9085] [Q&A]
I have a full viewport pixi app where I am trying to get some **pointerover** and **pointerout** events working for some sprites. As far as I can tell they are working as expected _UNLESS_ the mouse happens to be over the viewport when the application is loading (and doing a quick initial animation).  That is if I press the refresh button then quickly move the mouse over the viewport before everything is loaded/finished with the inital animation the **pointerover** and **pointerout** do not fire as expected. (the click event seems to still work as expected). If I wait until the initial loading/animation is complete, everything works fine. 

Also, if I simply move the mouse outside the browser viewport window then back in again the events resume working as expected. 

I tried manually simulating various  pointer events  to try and "reset" the pointer event system... but didn't find anything that worked.

Any ideas on what is going on or solutions I might try?

**Comment by bigtimebuddy** [comment]
This may be a bug? I tried to create a reproduce the issue you described, but couldn't. Could you provide an example (and some browser details) about how to repro this?

**Comment by batman4444** [comment]
Thanks for the response bigtimebuddy. I have not been able reproduce the issue outside of my project, which I am sure means it is something to do with my implementation. 

Any chance you know how to access _all_ the events that the pixi event system might be receiving?  
Also, is it possible to synthetically dispatch a pointer event to the pixi system?

**Reply by bigtimebuddy** [reply]
Here's an example of creating an event for pointermove. There are other methods like onPointerUp, etc. you can use to trigger other events.

```js

renderer.events.onPointerMove(
    new PointerEvent('pointermove', {
        clientX: 60,
        clientY: 60,
        pointerType: 'mouse',
    })
);
```
